### PR TITLE
Use proper colormap popup for the Colormap component

### LINKED ISSUE #6769: Colormap no longer previews in combox (Tensor view / DepthImage colormaps)
Before nice:
image

After not nice:
image

### PR DESCRIPTION
**Pull request by abey79**
### What

☝🏻 

- fixes #6769

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6771?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6771?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6771)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_viewer_context/src/gpu_bridge/colormap.rs [hunk 1]
```diff
@@ -64,40 +64,63 @@ fn colormap_preview_ui(
...
+                    }
+                }),
+        )
+    } else {
+        list_item.show_flat(ui, list_item::LabelContent::new(option.to_string()))
+    };
+
+    if response.clicked() {
+        *map = *option;
+    }
```
Comment (Wumpf) [inline crates/re_viewer_context/src/gpu_bridge/colormap.rs:92]: I think this is the culprit: clicked is the wrong thing to check for

**Review by Wumpf — CHANGES_REQUESTED** [review]
nice, but at least on the web demo here I can't select colormaps with this anymore.

File: crates/re_viewer_context/src/gpu_bridge/colormap.rs (hunk 1 above)
Comment (abey79) [reply crates/re_viewer_context/src/gpu_bridge/colormap.rs:92]: someone tested their PR properly it seems... 🙄👏🏻 🤦🏻 

Actually, `clicked()` is the right thing here, but the returning `Response` must be `marked_changed` to be picked up by the edit ui machinery. Fixed now.

**Review by Wumpf — APPROVED** [review]
ah makes sense, yes! Confirmed working now

**Pull request merged**